### PR TITLE
Breaking Change: Update Type exports

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
Type exports were hidden under factories. Now they are under `types`

Solution
========
Move the types to a new types.ts file and export it at the top level.

Double Checks:
---------------
- [x] Did you update the changelog?
- [x] Any new modules need to be exported?
- [x] Are new methods in the right module?
- [x] Do you have good documentation on exported methods?

Change summary:
---------------
* Updated all packages
* Updated typedoc config for the new version
* Updated formatting in accordance with new eslint rules
* Moved types to types.ts

Steps to Verify:
----------------
1. Build
1. Lint
1. Import and use

### CODE REVIEW

File: src/factories.ts [hunk 1]
```diff
@@ -1,191 +1,24 @@
-import { DSNPUserURI } from "./utilities/types";
 import { hash } from "./utilities/hash";
-
-interface ActivityContentBase {
```
Comment (wilwade) [inline src/factories.ts:4]: Moved these to `./types.ts`

File: src/factories.ts [hunk 2]
```diff
@@ -1,191 +1,24 @@
...
+  ActivityContentProfile,
+  ActivityContentVideo,
+  ActivityContentVideoLink,
+  DSNPUserURI,
+} from "./types";
 
 /**
  * createNote() provides a simple factory for generating an ActivityContentNote
  * object.
- *
```
Comment (wilwade) [inline src/factories.ts:188]: New formatting defaults

File: src/test/validation.test.ts [hunk 3]
```diff
@@ -805,7 +805,7 @@ describe("activity content validations", () => {
         it(`${testCase.name} throws correct error`, () => {
           expect(() =>
             requireIsActivityContentNoteType(testCase.testObject)
-          ).toThrowError("Invalid ActivityContent: " + testCase.expErr);
```
Comment (wilwade) [inline src/test/validation.test.ts:808]: New eslint standards

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
+interface ActivityContentBase {
```
Comment (wilwade) [inline src/types.ts:1]: Types moved to here.

File: typedoc.json [hunk 5]
```diff
@@ -1,12 +1,14 @@
 {
```
Comment (wilwade) [inline typedoc.json:1]: Formatting

**Review by enddynayn — APPROVED** [review]
💯 great!

**Review by JoeCap08055 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -83,7 +83,7 @@ jobs:
         run: npm run doc:json
 
       - name: Deploy 🚀
-        uses: JamesIves/github-pages-deploy-action@4.4.1
+        uses: JamesIves/github-pages-deploy-action@22a6ee251d6f13c6ab1ecb200d974f1a6feb1b8d
```
Comment (shannonwells) [inline .github/workflows/release.yml:86]: why is this using a specific commit?

**Review by shannonwells — APPROVED** [review]
Looks fine to me, just wondering if the specific commit is needed

File: .github/workflows/release.yml (hunk 6 above)
Comment (wilwade) [reply .github/workflows/release.yml:86]: It's the new version and we've been locking to specific commits (although after this was originally written.) Version tags can be altered by malicious actors, we don't have many of the versioned ones left, but this is one of them.

**Pull request merged**